rfd/0073-public-image-registry.md

```diff
@@ -0,0 +1,249 @@
+
+**What about name squatting on other container registry platforms?**
+
+A not yet finished RFD on third-party artifact mirroring will address concerns regarding name squatting on other container registry platforms. See [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2).
+
+### Scope
+
+This RFD will focus on the infrastructure, security, and observability of the replacement registry. It will also include a deprecation and migration plan for the existing Quay.io repositories.
+
+#### In Scope
+* Infrastructure plans with [example terraform](#appendix-a-example-terraform)
+* Migration and Deprecation plan
+
+#### Out of Scope
+* Image Signing w/ Cosign
+* Mirroring of images to third-party registries. See [Artifact Mirroring](https://github.com/gravitational/teleport/commit/2262efbb25463ccc135553d43293f6d8aee22ba2).
+
+### Infrastructure
+```
+             ┌─────────────────────────────────────────────────────────┐
+             │                                                         │
+             │    ┌───────────┐                   ┌──────────────────┐ │
+   Tag / Push│    │           │     Promotion     │                  │ │
+─────────────┼─►  │  AWS ECR  │  ──────────────►  │  AWS ECR Public  │ │◄───────── public.ecr.aws/gravitational/teleport
+             │    │           │                   │                  │ │
+             │    └───────────┘                   └──────────────────┘ │
+             │                                                         │
+             │AWS Account: teleport-prod                               │
+             └─────────────────────────────────────────────────────────┘
+```
+
+The infrastructure for this will live in the [cloud-terraform](https://github.com/gravitational/cloud-terraform) repository. The terraform for the `teleport-prod` account can be found [here](https://github.com/gravitational/cloud-terraform/tree/main/teleport-team/prod). Using AWS ECR and ECR Public allow us to rely on their managed infrastructure which reduces the operational complexity while enforcing our own security policies and allowing us to better audit changes to the environment. For more information on the pros and cons of alternatives, see [alternatives](#alternatives).
+
+### Security
+Amazon ECR and ECR Public have support for AWS IAM. With IAM we can create least-privileged policies that allow limited access to one or more part of the container-registry. For an example promotion user policy, see the [terraform example](#appendix-a-example-terraform). 
+
+In addition to AWS IAM, AWS supports our existing SSO infrastructure with Okta.
+
+As a part of observability, Cloudtrail logs will log all interactions with ECR which will allow the security team to create alerts for any changes to images. 
```

Comment:
"changes to images" >>  Can you discuss tag immutability as it does (or doesn't) apply to the public ecr repos?  I thought to some extent we'd be preventing changes to images in line with our "once a tag is published, it never changes" policy.

Reply:
Amazon ECR Public doesn't support tag immutability. This will be enforced in the first stage with Amazon ECR. You are correct that we would strive to have a "once a tag is published, it never changes" policy. However, RFD 0017 for artifact storage standards don't specify nor enforce that policy. It only states that administrative changes must be through an assumed role and are audited w/ alerts for any such modifications. 

For instance, an alert should be set up such that any write to the public ECR repository that doesn't come from the service account user should trigger an alert. It doesn't state that this can't happen, only that the security team should be alerted. 

I'm open to the idea of limiting even admin users from making changes even if they are audited. Would be cool if a role was available to make changes but was only ever given via short-lived credentials that are granted after being approved via a PR.

Reply:
> Amazon ECR Public doesn't support tag immutability. 

Gotcha.  I think this was the detail I was missing.  You might explicitly call this out as a limitation of Amazon ECR Public.